packages/sit-onyx/src/composables/checkAll.ts

```diff
@@ -0,0 +1,47 @@
+import type { OnyxCheckboxProps } from "src/components/OnyxCheckbox/types";
+import { computed, type Ref } from "vue";
+
+export type OptionValue = string | number | boolean;
+
+const useSelectAllCheckboxState = <TValue extends OptionValue = OptionValue>(
+  enabledOptionValues: Ref<TValue[]>,
+  modelValue: Ref<TValue[]>,
+) =>
+  computed<Partial<OnyxCheckboxProps>>(() => {
+    const currentValues = modelValue.value.filter((value) =>
+      enabledOptionValues.value.includes(value),
+    );
+    const isActive = currentValues.length === enabledOptionValues.value.length;
+
+    return {
+      modelValue: isActive,
+      indeterminate:
+        !isActive && enabledOptionValues.value.length && currentValues.length ? true : undefined,
+    };
+  });
```

Comment:
```suggestion
  computed(() => {
    const currentValues = modelValue.value.filter((value) =>
      enabledOptionValues.value.includes(value),
    );
    const isActive = currentValues.length === enabledOptionValues.value.length;

    return {
      modelValue: isActive,
      indeterminate:
        !isActive && enabledOptionValues.value.length && currentValues.length ? true : undefined,
    } satisfies Partial<OnyxCheckboxProps>;
  });
```

For better type inference :)